src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaInductionPersistenceAdapter.kt

```diff
@@ -7,35 +7,45 @@ import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.Ind
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.dto.CreateInductionDto
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.dto.UpdateInductionDto
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service.InductionPersistenceAdapter
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.PreviousQualificationsEntity
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.induction.InductionEntityMapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.induction.PreviousQualificationsEntityMapper
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.InductionRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.PreviousQualificationsRepository
```

Comment:
This class is where the lions share of the change is in respect of writing the PreviousQualifications when we create or update an Induction.

There is a little bit of logic that needs to happen which is all thoroughly tested in the unit test class. Some of it I've written "java style" with `if` statements as I felt the more condensed kotlin approach was harder to read. But happy to refactor this if there is a better approach 👍 

The logic is:

### When Creating an Induction
* If the prisoner has no qualifications already recorded (ie. before Induction) and the create induction DTO contains qualifications
  * Create the Induction and create the PreviousQualifications
* If the prisoner has qualifications already recorded (ie. before Induction) and the create induction DTO contains qualifications
  * Create the Induction and update the PreviousQualifications
* If the prisoner does not have qualifications already recorded (ie. before Induction) and the create induction DTO does not contain qualifications
  * Create the Induction and do not create PreviousQualifications
* If the prisoner has qualifications already recorded (ie. before Induction) and the create induction DTO does not contain qualifications
  * Create the Induction and delete the existing PreviousQualifications

### When Updating an Induction
* If the prisoner has no qualifications already recorded and the update induction DTO contains qualifications
  * Update the Induction and create the PreviousQualifications
* If the prisoner has qualifications already recorded and the create induction DTO contains qualifications
  * Update the Induction and update the PreviousQualifications
* If the prisoner does not have qualifications already recorded and the create induction DTO does not contain qualifications
  * Update the Induction and do not create or update PreviousQualifications
* If the prisoner has qualifications already recorded and the create induction DTO does not contain qualifications
  * Create the Induction and do not update PreviousQualifications

This last scenario may sound surprising but is specifically mentioned in the swagger spec for updating an Induction:
> If a field (or "child" object) is provided in the request, it will completely overwrite the existing value. Conversely, if a field is not included in the request, then the existing value will remain untouched (including its updatedAt/updatedBy values).
And there is an integration test already in place that proves this functionality.

Reply:
When the new standalone flow for qualifications is added I assume that can be used and we can deprecate previous quals from the update induction flow?